src/libpmemkv.cc

```diff
@@ -46,6 +46,74 @@
 #include "engines-experimental/stree.h"
 #include "engines-experimental/caching.h"
 #endif
+#include <unordered_map>
+
+typedef struct pmemkv_config_value pmemkv_config_value;
+typedef std::unordered_map<std::string, pmemkv_config_value> umap_t;
+
+struct pmemkv_config_value {
+	char *buffer;
+	int32_t size;
+};
+
+struct pmemkv_config {
+	umap_t umap;
+};
+
+pmemkv_config*
+pmemkv_config_new(void)
+{
+	return new pmemkv_config;
+}
+
+void
+pmemkv_config_delete(pmemkv_config* config)
+{
+	for (auto element : config->umap)
+		delete[] element.second.buffer;
+
+	delete config;
+}
+
+int
+pmemkv_config_put(pmemkv_config* config, const char* key,
+			const void* value, size_t value_len)
+{
+	std::string mkey(key);
+
+	pmemkv_config_value mvalue;
+	mvalue.size = value_len;
+	mvalue.buffer = new char [mvalue.size];
+	memcpy(mvalue.buffer, value, mvalue.size);
+
+	config->umap[mkey] = mvalue;
```

Comment:
@igchor correct me if I'm wrong, but I think it's more optimal to use the insert method. The way you implemented this it will call the default constructor and operator=, instead of just calling copy constructor.